### PR TITLE
Chore: Convert Pagination (v2) to Typescript

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

Converts `v2/Pagination` to Typescript.

### Why is it needed?

https://github.com/strapi/design-system/issues/983

### CODE REVIEW

File: packages/strapi-design-system/src/v2/Pagination/components.tsx [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
 import { buttonFocusStyle } from '../../themes/utils';
 import { Typography } from '../../Typography';
 import { VisuallyHidden } from '../../VisuallyHidden';
 
-const PaginationText = styled(Typography)`
-  line-height: revert;
-`;
+interface PaginationLinkProps extends BaseLinkProps {
+  active: boolean;
+  children: React.ReactNode;
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/v2/Pagination/components.tsx:15]:
```suggestion
```

not sure you need this as `BaseLinkProps` already has children.

File: packages/strapi-design-system/src/v2/Pagination/components.tsx [hunk 2]
```diff
@@ -77,9 +81,7 @@ export const PreviousLink = React.forwardRef(({ children, ...props }, ref) => {
   );
 });
 
-PreviousLink.displayName = 'PreviousLink';
-
-export const NextLink = React.forwardRef(({ children, ...props }, ref) => {
+export const NextLink = React.forwardRef<HTMLElement, PaginationLinkProps>(({ children, ...props }, ref) => {
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/v2/Pagination/components.tsx:84]:
```suggestion
export const NextLink = React.forwardRef<HTMLAnchorElement, PaginationLinkProps>(({ children, ...props }, ref) => {
```

File: packages/strapi-design-system/src/v2/Pagination/components.tsx [hunk 3]
```diff
@@ -92,46 +94,28 @@ export const NextLink = React.forwardRef(({ children, ...props }, ref) => {
...
-      <VisuallyHidden>{children}</VisuallyHidden>
-      <PaginationText aria-hidden variant="pi" fontWeight={isActive ? 'bold' : null}>
-        {number}
-      </PaginationText>
-    </PageLinkWrapper>
-  );
-});
-
-PageLink.displayName = 'PageLink';
+export const PageLink = React.forwardRef<HTMLElement, PaginationPageLinkProps>(
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/v2/Pagination/components.tsx:97]:
```suggestion
export const PageLink = React.forwardRef<HTMLAnchorElement, PaginationPageLinkProps>(
```

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Minor tweaks, looks good otherwise 👍🏼

File: packages/strapi-design-system/src/v2/Pagination/components.tsx [hunk 4]
```diff
@@ -60,11 +68,7 @@ const ActionLinkWrapper = styled(LinkWrapper)`
       : undefined}
 `;
 
-const DotsWrapper = styled(LinkWrapper)`
-  color: ${({ theme }) => theme.colors.neutral800};
-`;
-
-export const PreviousLink = React.forwardRef(({ children, ...props }, ref) => {
+export const PreviousLink = React.forwardRef<HTMLElement, PaginationLinkProps>(({ children, ...props }, ref) => {
```
[the remark this answers is not in the document]
Comment (gu-stav) [reply packages/strapi-design-system/src/v2/Pagination/components.tsx:71]: Good idea to make them more specific.

**Review by joshuaellis — APPROVED** [review]
nice work!

**Pull request merged**